File: frontend/src/mappers/Checklist.ts
import { Checklist, ChecklistDto } from "../interfaces";

export const mapChecklistToChecklistDto = (
  checklist: Checklist
): ChecklistDto => {
  return {
    dueDate: checklist.dueDate,
    numContrato: checklist.numContrato,
    numDestaLV: checklist.numDestaLV,
    numDocumento: checklist.numDocumento,
    projeto: checklist.projeto,
    revisao: checklist.revisao,
    titutlo: checklist.titutlo,
    verificador: checklist.verificador,
    executante: checklist.executante,
    anteriorId: checklist.anteriorId,
    caminho: checklist.caminho,
    emitido: checklist.emitido,
    paraVerificar: checklist.paraVerificar,
    versao: checklist.versao,
  };
};

File: api/Models/Checklist.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace api.Models
{
    [Table("Checklist")]
    public class Checklist
    {
        public int Id { get; set; }
        public required string NumDestaLV { get; set; }
        public required string NumContrato { get; set; }
        public required string NumDocumento { get; set; }
        public required string Projeto { get; set; }
        public required int Revisao { get; set; }
        public required string Titutlo { get; set; }
        public required string Verificador { get; set; }
        public required string Executante { get; set; }
        public required int Versao { get; set; }
        public required int AnteriorId { get; set; }
        public required bool ParaVerificar { get; set; }
        public required bool Emitido { get; set; }
        public required string Caminho { get; set; }

        public DateTime CreatedOn { get; set; } = DateTime.Now;
        public DateTime DueDate { get; set; }
        public List<Check> Checks { get; set; } = new List<Check>();

        public required string AppUserIdCord { get; set; }
        public AppUser AppUserCord { get; set; }
        public required string AppUserIdExec { get; set; }
        public AppUser AppUserExec { get; set; }
    }
}